lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -3,10 +3,11 @@ use std::ops::Range;
 use geo::algorithm::haversine_distance::HaversineDistance;
 #[allow(deprecated)]
 use geo::{Coordinate, Point};
+use geo::{Intersects, Polygon};
 use geohash::{decode, decode_bbox, encode, Direction, GeohashError};
 use itertools::Itertools;
 
-use crate::types::{GeoBoundingBox, GeoPoint, GeoRadius};
+use crate::types::{GeoBoundingBox, GeoPoint, GeoPolygon, GeoRadius};
 
 pub type GeoHash = String;
```

Comment:
What about using a bytestring here (`Vec<u8>` and `&[u8]`)?

These are usually quite a bit faster on hardware due to better optimizations. What do you think of this, or do you have a strong objection against it?

I know this isn't the PR where it was introduced, but it's just a thought that came up during this review.

Reply:
Ack. Good point. I can test about bytestring later.